src/hotspot/share/opto/type.cpp

```diff
@@ -2575,8 +2575,8 @@ bool TypeVect::eq(const Type *t) const {
 
 //------------------------------hash-------------------------------------------
 // Type-specific hashing function.
-int TypeVect::hash(void) const {
-  return (intptr_t)_elem + (intptr_t)_length;
+uint TypeVect::hash(void) const {
+  return (uint)(uintptr_t)_elem + (uint)(uintptr_t)_length;
```

Comment:
I did it this way on purpose, because it makes more sense to me to do the `+` as `uint` rather than `uintptr_t`.  The `uintptr_t` is there only because the fields are pointers, but the hash function doesn't need to do 64-bit `+` on a 64-bit platform, though I guess it wouldn't hurt either.